scipy/signal/sigtoolsmodule.c

```diff
@@ -762,15 +770,17 @@ static int pre_remez(double *h2, int numtaps, int numbands, double *bands, doubl
 /* N-D Order Filtering. */
 
 
-static void fill_buffer(char *ip1, PyArrayObject *ap1, PyArrayObject *ap2, char *sort_buffer, int nels2, int check, intp *loop_ind, intp *temp_ind, uintp *offset){ 
+static void fill_buffer(char *ip1, PyArrayObject *ap1, PyArrayObject *ap2,
+                        char *sort_buffer, int nels2, int check,
+                        intp *loop_ind, intp *temp_ind, uintp *offset){ 
   int i, k, incr = 1;
-  int ndims = ap1->nd;
-  intp *dims2 = ap2->dimensions;
-  intp *dims1 = ap1->dimensions;
-  intp is1 = ap1->descr->elsize;
-  intp is2 = ap2->descr->elsize;
-  char *ip2 = ap2->data;
-  int elsize = ap1->descr->elsize;
+  int ndims = PyArray_NDIM(ap1);
+  intp *dims2 = PyArray_DIMS(ap2);
+  intp *dims1 = PyArray_DIMS(ap1);
+  intp is1 = PyArray_DESCR(ap1)->elsize;
```

Comment:
This (and other references to `->descr->elsize` elsewhere) could be written 

    intp is1 = PyArray_ITEMSIZE(ap1);